packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -840,6 +840,16 @@ class ThemeData with Diagnosticable {
   factory ThemeData.dark({bool? useMaterial3}) =>
       ThemeData(brightness: Brightness.dark, useMaterial3: useMaterial3);
 
+  factory ThemeData.highContrastLight() => ThemeData.from(
```

Comment:
The default colors in `ColorScheme.highContrastLight()` is still M2. Instead of using ColorScheme.highContrastLight(), I think we can generate:

```dart
useMaterial3 
? ColorScheme.fromSeed(seedColor: Color(0xFF6750A4), contrastLevel: 1.0),
: ColorScheme.highContrastLight(),
```

`Color(0xFF6750A4)` is the primary color for M3 base colors. Using it as the color seed can generate a similar(almost the same) color scheme with the default m3 colors. Changing the contrast level to 1.0 will generate a high-contrast color scheme.

Then maybe we can apply the light forced color here?

Reply:
From the latest M3 website, we don't have any specs to specify the static high-contrast colors as the base colors, so I think the only way to generate a high-contrast color scheme that matches the M3 style is by providing a similar seed color and setting the contrast level to 1.0

Reply:
If we have to override fromseed to use systemcolor, we can add a flag to the fromseed, maybe `fromSeed(useSystemColor: true)`, if it is set to true, it will override colors over the generated one. WDYT?

Reply:
@chunhtai what happens to the seed color when `useSystemColors` is set to true?

The way I think about it is seed color is mutually exclusive with system colors. Mixing those two is tricky and probably undesirable IMO.

Reply:
> Doing ColorScheme.fromSeed(seedColor: Color(0xFF6750A4), contrastLevel: 1.0) seems against the goal of this PR. What do you think is the best way to use forced system colors with M3?

After we generate the whole color scheme by using `fromSeed`, we can override the according colors using forced colors by `copyWith()`. Does this make sense? Similar to how we override the `highContrastLight` in this PR.

Reply:
> After we generate the whole color scheme by using `fromSeed`, we can override the according colors using forced colors by `copyWith()`. Does this make sense? Similar to how we override the `highContrastLight` in this PR.

That makes sense. I'm wary of mixing colors from different sources in case they don't match well. But I'm not against trying it and seeing what happens. We can iterate on it if we receive feedback from users.

Reply:
Yes! And I think it should be fine as long as we map them correctly:)

Reply:
> I'm wary of mixing colors from different sources in case they don't match well

I think this is fine, it is user's choice for the color. It is their explicitly choice, and I think we should respect that as much as possible. IMO, I think it be worse if they explicit pick a color and the color doesn't affect the UI.